impf_termin_tester/utils/send_mail.py

```diff
@@ -0,0 +1,131 @@
+
+    if not to and not cc and not bcc:
+        return False, "No recipients, nothing to do"
+
+    # Create the message
+    msg = EmailMessage()
+
+    msg.set_content(text)
+    if html:
+        msg.add_alternative(html, subtype='html')
+
+    msg['From'] = mail_from
+    if to:
+        msg['To'] = to
+    if cc:
+        msg['CC'] = cc
+    if bcc:
+        msg['BCC'] = bcc
+    msg['Subject'] = subject
+    msg['Date'] = formatdate()
+    msg['Message-ID'] = make_msgid()
+    msg['Auto-Submitted'] = 'auto-generated'  # RFC 3834 section 5
+
+    # Send the message
+    try:
+        logging.debug("trying to send mail (smtp) via smtp server '{0}'".format(mail_server))
+        host, port = (mail_server + ':25').split(':')[:2]
+        server = smtplib.SMTP(host, int(port))
+        try:
+            server.ehlo()
+            try:  # try to do TLS
+                if server.has_extn('starttls'):
+                    server.starttls()
+                    server.ehlo()
+                    logging.debug("tls connection to smtp server established")
+            except Exception:
+                logging.debug("could not establish a tls connection to smtp server, continuing without tls")
+            # server.set_debuglevel(1)
+            if login_user is not None and login_password is not None:
+                logging.debug("trying to log in to smtp server using account '{0}'".format(login_user))
+                server.login(login_user, login_password)
+            server.send_message(msg)
+        finally:
+            try:
+                server.quit()
+            except AttributeError:
+                # in case the connection failed, SMTP has no "sock" attribute
+                pass
+    except smtplib.SMTPException as e:
+        logging.exception("smtp mail failed with an exception.")
+        return False, str(e)
+    except (os.error, socket.error) as e:
+        logging.exception("smtp mail failed with an exception.")
+        return False, "Connection to mailserver '%s' failed: %s" % (mail_server, str(e))
+
+    logging.debug("Mail sent successfully")
+    return True, "Mail sent successfully"
+
+
+if __name__ == '__main__':
+    # some testing code you can edit to play with this code by calling it directly like: python3 send_mail.py
+    me = '"Joe Doe-Sender" <joe@example.org>'
+    them = '"Jane Doe-Receiver" <jane@example.org>'
+    # note: if sending to many people, respect their privacy and put them ALL into BCC,
+    #       so you do not disclose all their email addresses to all of them!
+    #       of course, you need to put also something into TO - just use same address as FROM for that.
+    ok, msg = send_mail(mail_from=me, to=[me, ], bcc=[them, them, them, ],
+                        subject="Test send_mail", text="Test!",
+                        mail_server="mail.example.org:587", login_user="username", login_password="password")
+    print("ok: %r, msg: %r" % (ok, msg))
+
```

Comment:
Do you wanna leave this in here as some kind of demo code? feels a bit detached from the project itself.

Reply:
Yes, this is testing code. It won't disturb anything when importing this as a library, due to the if in line 120.

But it serves as an example, as testing code, and as useful hint how not to send mass mails, so please keep it.